src/main/resources/templates/signup.html

```diff
@@ -0,0 +1,100 @@
+                               th:field="*{lastname}" th:placeholder="#{signup.input.lastname}" type="text">
+                        <span class="text-warning" th:if="${invalidLastNameLength}"
+                              th:text="#{signup.msg.lastname}"></span>
+                    </label>
+                </div>
+            </div>
+            <div class="row mb-3">
+                <div class="col-sm-6 mb-3 mb-sm-0">
+                    <label>
+                        <span th:text="#{signup.label.ssn}">Enter social security number</span>
+                        <input class="form-control" name="ssn" placeholder="YYMMDD-XXXX" required
+                               th:field="*{ssn}" th:placeholder="#{signup.input.ssn}" type="text">
+                        <span class="text-warning" th:if="${invalidSSN}"
+                              th:text="#{signup.msg.ssn}"></span>
+                    </label>
+                </div>
+            </div>
+            <div class="row mb-3">
+                <div class="col-sm-6 mb-3 mb-sm-0">
+                    <label>
+                        <span th:text="#{signup.label.email}">Enter email</span>
+                        <input class="form-control" name="email" placeholder="Email" required
+                               th:field="*{email}" th:placeholder="#{signup.input.email}" type="email">
+                        <span class="text-warning" th:if="${invalidEmail}"
+                              th:text="#{signup.msg.email}"></span>
+                        <span class="text-warning" th:if="${emailIsTaken}" th:text="#{signup.msg.email.taken}"></span>
+                    </label>
+                </div>
+            </div>
+            <div class="row mb-3">
+                <div class="col-sm-6 mb-3 mb-sm-0">
+                    <label>
+                        <span th:text="#{signup.label.username}">Enter username</span>
+                        <input class="form-control" name="username" placeholder="Username" required
+                               th:field="*{username}" th:placeholder="#{signup.input.username}" type="text">
+                        <span class="text-warning" th:if="${invalidUsernameLength}"
+                              th:text="#{signup.msg.username}"></span>
+                        <span class="text-warning" th:if="${usernameIsTaken}"
+                              th:text="#{signup.msg.username.taken}"></span>
+                    </label>
+                </div>
+            </div>
+            <div class="row mb-3">
+                <div class="col-sm-6 mb-3 mb-sm-0">
+                    <label>
+                        <span th:text="#{signup.label.password}">Enter password</span>
+                        <input class="form-control" name="password" placeholder="Password" required
+                               th:field="*{password}" th:placeholder="#{signup.input.password}" type="password">
```

Comment:
Add this: 
pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}" th:title="#{signup.label.passwordMessage}"

And add this to the language config:
signup.label.passwordMessage: "Must contain at least one number and one uppercase and lowercase letter, and at least 8 or more characters"

signup.label.passwordMessage: "Måste innehålla minst ett nummer samt en store bokstav och liten bokstav. Lösenordet måste även innehålla minst 8 tecken.